lib/spack/spack/environment.py

```diff
@@ -553,6 +625,13 @@ def remove(self, query_spec, force=False):
                 del self.concretized_order[i]
                 del self.specs_by_hash[dag_hash]
 
+        index = list(self.read_specs.keys()).index(list_name)
```

Comment:
I agree this is more straightforward than before

I think it could use a comment though: something like "read_specs is an ordered dict, all list entries after the modified list may refer to it and that reference may now be stale, so they should be updated"